cgi/product_multilingual.pl

```diff
@@ -1148,425 +1163,170 @@ ($product_ref, $field, $language, $request_ref)
 		my $class = 'main';
 		my $prefix = '';
 
-		my $shown = 0;
+		my $nutrient_shown = 0;
 
+		# In the nutrient table, nutrients ending with a - minus sign are not shown by default
 		if (
-			   ($nutriment !~ /-$/)
-			or ((defined $product_ref->{nutriments}{$nid}) and ($product_ref->{nutriments}{$nid} ne ''))
-			or (    (defined $product_ref->{nutriments}{$nid . "_prepared"})
-				and ($product_ref->{nutriments}{$nid . "_prepared"} ne ''))
-			or (    (defined $product_ref->{nutriments}{$nid . "_modifier"})
-				and ($product_ref->{nutriments}{$nid . "_modifier"} eq '-'))
-			or (    (defined $product_ref->{nutriments}{$nid . "_prepared_modifier"})
-				and ($product_ref->{nutriments}{$nid . "_prepared_modifier"} eq '-'))
-			or ($nid eq 'new_0')
-			or ($nid eq 'new_1')
+			($nutrient !~ /-$/)
+			# FIXME: add an or condition that is true if we have a value or modifier in any of the input sets for the selected source
+			or ($nid eq 'new_0') or ($nid eq 'new_1')
```

Comment:
I think new_0 and new_1 do not exist anymore, do they ? (because I don't see them in the foeach loop)

Reply:
Good catch, removing them